cpp/oneapi/dal/algo/covariance/backend/gpu/compute_kernel_dense_dpc.cpp

```diff
@@ -0,0 +1,135 @@
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "daal/src/algorithms/covariance/oneapi/covariance_kernel_oneapi.h"
+
+#include "oneapi/dal/algo/covariance/backend/gpu/compute_kernel.hpp"
+#include "oneapi/dal/backend/interop/common_dpc.hpp"
+#include "oneapi/dal/backend/interop/error_converter.hpp"
+#include "oneapi/dal/backend/interop/table_conversion.hpp"
+
+#include "oneapi/dal/table/row_accessor.hpp"
+
+namespace oneapi::dal::covariance::backend {
+
+using dal::backend::context_gpu;
+using descriptor_t = detail::descriptor_base<task::compute>;
+
+namespace daal_covariance = daal::algorithms::covariance;
+namespace interop = dal::backend::interop;
+
+template <typename Float>
+using daal_covariance_kernel_t = daal_covariance::oneapi::internal::
+    CovarianceDenseBatchKernelOneAPI<Float, daal_covariance::Method::defaultDense>;
+
+template <typename Float, typename Task>
+static compute_result<Task> call_daal_kernel(const context_gpu& ctx,
+                                             const descriptor_t& desc,
+                                             const table& data) {
+    bool is_mean_computed = false;
+
+    auto& queue = ctx.get_queue();
+    interop::execution_context_guard guard(queue);
+
+    const std::int64_t component_count = data.get_column_count();
+
+    daal_covariance::Parameter daal_parameter;
+    daal_parameter.outputMatrixType = daal_covariance::covarianceMatrix;
+
+    dal::detail::check_mul_overflow(component_count, component_count);
+
+    const auto daal_data = interop::convert_to_daal_table(queue, data);
+
+    auto arr_means = array<Float>::empty(queue, component_count, sycl::usm::alloc::device);
+    const auto daal_means = interop::convert_to_daal_table(queue, arr_means, 1, component_count);
+
+    auto result = compute_result<Task>{}.set_result_options(desc.get_result_options());
+
+    if (desc.get_result_options().test(result_options::cov_matrix)) {
+        auto arr_cov_matrix =
+            array<Float>::empty(queue, component_count * component_count, sycl::usm::alloc::device);
+        const auto daal_cov_matrix =
+            interop::convert_to_daal_table(queue, arr_cov_matrix, component_count, component_count);
+
+        interop::status_to_exception(
+            daal_covariance_kernel_t<Float>().compute(daal_data.get(),
+                                                      daal_cov_matrix.get(),
+                                                      daal_means.get(),
+                                                      &daal_parameter));
+        is_mean_computed = true;
+
+        result.set_cov_matrix(
+            homogen_table::wrap(arr_cov_matrix, component_count, component_count));
+    }
+    if (desc.get_result_options().test(result_options::cor_matrix)) {
```

Comment:
Why are you copied much code when the different is only in the parameter value `daal_parameter.outputMatrixType`?